### ISSUE TITLE
Ambiguous code

### ISSUE THREAD

**Issue by ArlindKadra** [automl/Auto-PyTorch#277]
When calling `predict `from `base_pipeline`:
https://github.com/automl/Auto-PyTorch/blob/3413bc32758b4be47b4511a09b64aff4f3a8e005/autoPyTorch/pipeline/base_pipeline.py#L167-L185

`get_loader` is also called, however, get loader only applies the test transformations, no matter what set we are using:
https://github.com/automl/Auto-PyTorch/blob/3413bc32758b4be47b4511a09b64aff4f3a8e005/autoPyTorch/pipeline/components/training/data_loader/base_data_loader.py#L141-L159

Overall, this does not seem like the right behavior. You could also call predict with the train or validation set to get more information since and like this you always apply the test transformations. `get_loader `does not seem to be implemented properly.
